fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -144,125 +150,45 @@ public CompletableFuture<RecordCursorResult<IndexEntry>> onNext() {
-                    }
-                    if (timer != null) {
-                        timer.increment(FDBStoreTimer.Counts.LOAD_SCAN_ENTRY);
-                    }
-                    if (limitRemaining != Integer.MAX_VALUE) {
-                        limitRemaining--;
-                    }
-                    Tuple setPrimaryKey = Tuple.fromBytes(pk.bytes);
-                    // Initialized with values that aren't really legal in a Tuple to find offset bugs.
-                    List<Object> fieldValues = Lists.newArrayList(fields);
-                    if (groupingKey != null) {
-                        // Grouping keys are at the front.
-                        for (int i = 0; i < groupingKey.size(); i++) {
-                            fieldValues.set(i, groupingKey.get(i));
-                        }
-                    }
-                    if (storedFields != null) {
-                        for (int i = 0; i < storedFields.size(); i++) {
-                            if (storedFieldTypes.get(i) == null) {
-                                continue;
-                            }
-                            Object value = null;
-                            IndexableField docField = document.getField(storedFields.get(i));
-                            switch (storedFieldTypes.get(i)) {
-                                case STRING:
-                                    value = docField.stringValue();
-                                    break;
-                                case BOOLEAN:
-                                    value = Boolean.valueOf(docField.stringValue());
-                                    break;
-                                case INT:
-                                case LONG:
-                                case DOUBLE:
-                                    value = docField.numericValue();
-                                    break;
-                                default:
-                                    break;
-                            }
-                            fieldValues.set(i, value);
-                        }
-                    }
-                    int[] keyPos = state.index.getPrimaryKeyComponentPositions();
-                    Tuple tuple;
-                    if (keyPos != null) {
-                        List<Object> leftovers = Lists.newArrayList();
-                        for (int i = 0; i < keyPos.length; i++) {
-                            if (keyPos[i] > -1) {
-                                fieldValues.set(keyPos[i], setPrimaryKey.get(i));
-                            } else {
-                                leftovers.add(setPrimaryKey.get(i));
-                            }
-                        }
-                        tuple = Tuple.fromList(fieldValues).addAll(leftovers);
-                    } else {
-                        tuple = Tuple.fromList(fieldValues).addAll(setPrimaryKey);
-                    }
+            skip -= pageSize;
```

Comment:
Probably should reduce by _actual_ number of results that `searchAfter` has advanced.

Reply:
Since this is only to handle the pages which fall completely into skip range. this should be fine. The first page which has both records to skip and records to return, will be handled by ```maybePerformScan();``` below this.